src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -720,6 +720,9 @@ object RefChecks {
           case List(param) =>
             if (param.is(Mutable))
               ctx.error("value class parameter must not be a var", param.pos)
+
+            if (isDerivedValueClass(param.info.typeSymbol))
+              ctx.error("value class may not wrap another user-defined value class", param.pos)
```

Comment:
No. That's completely fine and supported in Dotty. The only issue is when there is a cycle, this happens a value class underlying type (or the underlying type of its underlying type or ...) is itself. There is a `checkNonCyclic` method in `ExtensionMethods` that is supposed to do this but is currently disabled, try enabling it:
https://github.com/lampepfl/dotty/blob/master/src/dotty/tools/dotc/transform/ExtensionMethods.scala#L148-L150